agent/src/main/scala/za/co/absa/atum/agent/model/AtumPartitions.scala

```diff
@@ -0,0 +1,54 @@
+package za.co.absa.atum.agent.model
+
+import za.co.absa.atum.agent.model.AtumPartitions.Partitions
+
+import java.time.LocalDate
+import scala.collection.immutable.ListMap
+
+case class AtumPartitions(name: String, reportDate: LocalDate, partitions: Partitions = ListMap()) {
```

Comment:
This class is kind of a key toe `AtumContext` definitely should be a member of the class.